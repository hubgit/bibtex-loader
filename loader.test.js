const loader = require('./loader')

test('Converts BibTeX to CSL JSON', () => {
  const input = `
@inproceedings{saad2013supporting,
  title={Supporting literature review by searching, visualizing and navigating related papers},
  author={Saad, Farag and Mathiak, Brigitte and Mutschke, Peter},
  booktitle={2013 International Conference on Cloud and Green Computing},
  pages={363--368},
  year={2013},
  organization={IEEE}
}

@article{qiao2013cociter,
  title={CoCiter: an efficient tool to infer gene function by assessing the significance of literature co-citation},
  author={Qiao, Nan and Huang, Yi and Naveed, Hammad and Green, Christopher D and Han, Jing-Dong J},
  journal={PloS one},
  volume={8},
  number={9},
  pages={e74074},
  year={2013},
  publisher={Public Library of Science}
}

@inproceedings{rind2017pubviz,
  title={PubViz: Lightweight visual presentation of publication data},
  author={Rind, A and Haberson, A and Blumenstein, K and Niederer, C and Wagner, M and Aigner, W},
  booktitle={Proc. Eurographics Conf. Visualization (EuroVis)--Short Paper. Accepted},
  year={2017}
}`

  const output = loader(input)

  expect(output).toMatchSnapshot()
})
