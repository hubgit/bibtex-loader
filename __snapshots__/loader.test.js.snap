// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converts BibTeX to CSL JSON 1`] = `"module.exports = [{\\"title\\":\\"Supporting literature review by searching, visualizing and navigating related papers\\",\\"author\\":[{\\"given\\":\\"Farag\\",\\"family\\":\\"Saad\\"},{\\"given\\":\\"Brigitte\\",\\"family\\":\\"Mathiak\\"},{\\"given\\":\\"Peter\\",\\"family\\":\\"Mutschke\\"}],\\"container-title\\":\\"2013 International Conference on Cloud and Green Computing\\",\\"page\\":\\"363-368\\",\\"type\\":\\"paper-conference\\",\\"citation-label\\":\\"saad2013supporting\\",\\"id\\":\\"saad2013supporting\\",\\"year-suffix\\":\\"supporting\\",\\"issued\\":{\\"date-parts\\":[[2013]]}},{\\"title\\":\\"CoCiter: an efficient tool to infer gene function by assessing the significance of literature co-citation\\",\\"author\\":[{\\"given\\":\\"Nan\\",\\"family\\":\\"Qiao\\"},{\\"given\\":\\"Yi\\",\\"family\\":\\"Huang\\"},{\\"given\\":\\"Hammad\\",\\"family\\":\\"Naveed\\"},{\\"given\\":\\"Christopher D\\",\\"family\\":\\"Green\\"},{\\"given\\":\\"Jing-Dong J\\",\\"family\\":\\"Han\\"}],\\"container-title\\":\\"PloS one\\",\\"volume\\":\\"8\\",\\"issue\\":\\"9\\",\\"page\\":\\"e74074\\",\\"publisher\\":\\"Public Library of Science\\",\\"type\\":\\"article-journal\\",\\"citation-label\\":\\"qiao2013cociter\\",\\"id\\":\\"qiao2013cociter\\",\\"year-suffix\\":\\"cociter\\",\\"issued\\":{\\"date-parts\\":[[2013]]}},{\\"title\\":\\"PubViz: Lightweight visual presentation of publication data\\",\\"author\\":[{\\"given\\":\\"A\\",\\"family\\":\\"Rind\\"},{\\"given\\":\\"A\\",\\"family\\":\\"Haberson\\"},{\\"given\\":\\"K\\",\\"family\\":\\"Blumenstein\\"},{\\"given\\":\\"C\\",\\"family\\":\\"Niederer\\"},{\\"given\\":\\"M\\",\\"family\\":\\"Wagner\\"},{\\"given\\":\\"W\\",\\"family\\":\\"Aigner\\"}],\\"container-title\\":\\"Proc. Eurographics Conf. Visualization (EuroVis)â€“Short Paper. Accepted\\",\\"type\\":\\"paper-conference\\",\\"citation-label\\":\\"rind2017pubviz\\",\\"id\\":\\"rind2017pubviz\\",\\"year-suffix\\":\\"pubviz\\",\\"issued\\":{\\"date-parts\\":[[2017]]}}]"`;
